bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/util/AsyncBackOff.java

```diff
@@ -0,0 +1,83 @@
+package com.google.cloud.bigtable.util;
+
+import com.google.api.client.util.BackOff;
+import com.google.api.client.util.ExponentialBackOff;
+import com.google.cloud.bigtable.config.Logger;
+import com.google.cloud.bigtable.grpc.io.RefreshingOAuth2CredentialsInterceptor;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Stopwatch;
+import com.google.common.base.Ticker;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Wrapper around {@link ExponentialBackOff} to allow it to be used w/o a sleeper.
+ * Its marriage of a rate limiter and an ExponentialBackOff.
+ * This class is not threadsafe.
+ */
+public class AsyncBackOff {
+  private static final Logger LOG = new Logger(AsyncBackOff.class);
+
+  private final Stopwatch stopwatch;
+  private final BackOff backOff;
+  private long currentWait = 0;
+
+  public AsyncBackOff() {
+    this(5, 5000);
+
+  }
+  public AsyncBackOff(int initialIntervalMillis, int maxIntervalMillis) {
+    stopwatch = Stopwatch.createUnstarted();
+    backOff = new ExponentialBackOff.Builder()
```

Comment:
I'm not sure this will work.  ExponentialBackOff's constructor starts a timer.  When maxIntervalMillis + start hits, backoff consistently returns Backoff.STOP.  I think that the Backoff should only be created in `tryAcquire`, and should have a default value of null.